src/ripple/overlay/impl/PeerImp.cpp

```diff
@@ -2083,6 +2083,14 @@ PeerImp::getLedger (std::shared_ptr<protocol::TMGetLedger> const& m)
     p_journal_.trace <<
         "GetLeder: " << logMe;
 
+    int depth = -1;
+    if (packet.has_querydepth())
+        depth = packet.querydepth();
```

Comment:
Upon reviewing the code it looks to me like `query depth()` can't return negative.  It would be less confusing if the code reflected that invariant.  Maybe something like:
````
    int depth;
    if (packet.has_querydepth())
        depth = std::min(packet.querydepth(), 3);
    else
        depth = isHighLatency() ? 2 : 1;
````